src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2076,9 +2115,18 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             TSFInputControl().Close();
             _autoScrollTimer.Stop();
 
-            _interactivity.Close();
+            if (!_detached)
+            {
+                _interactivity.Close();
+            }
         }
     }
+    void TermControl::Detach()
+    {
+        _revokers = {};
+        _interactivity.Detach();

```

Comment:
important to note that we still _have_ an `_interactivity` member that is fully populated after this, so there is technically a state where it thinks it's detached but still sends it messages...

Reply:
Tempted to replace this with 

```c++

        _revokers = {};
        Control::ControlInteractivity old{ nullptr };
        std::swap(old, _interactivity);
        old.Detach();
        _detached = true;
```

but I don't have time to validate tonight.